signer/src/transaction_signer.rs

```diff
@@ -626,22 +603,21 @@ where
                 )
                 .increment(1);
 
-                if !self.wsts_state_machines.contains_key(&msg.txid) {
-                    let (maybe_aggregate_key, _) = self
-                        .get_signer_set_and_aggregate_key(bitcoin_chain_tip)
-                        .await?;
-
-                    let state_machine = SignerStateMachine::load(
-                        &db,
-                        maybe_aggregate_key.ok_or(Error::NoDkgShares)?,
-                        self.threshold,
-                        self.signer_private_key,
-                    )
+                let id = validation_sighash?.into();
```

Comment:
heads-up/FYI/maybe this will bite us in the future: previously with the `contains_key` we would call `get_signer_set_and_aggregate_key` only once per bitcoin tx, while now we will call it once per input, and since that one calls `get_last_key_rotation` we may increase the signing round time by a couple of seconds (or more) until we address that.

Reply:
In https://github.com/stacks-network/sbtc/pull/1210 I replace that look-up using `get_signer_set_and_aggregate_key` with a look-up using the `bitcoin_tx_sighashes.sighash` column. That should be much faster, since the `sighash` column is the primary key for that table.